app/controllers/api/v1/subscriptions_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Api::V1::SubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    if params[:stripeToken]
+      begin
+        customer =
+          Stripe::Customer.create(
+            email: current_user.email, source: params[:stripeToken]
+          )
+  
+        charge =
+          Stripe::Charge.create(
+            customer: customer.id,
+            amount: 10_000,
+            description: 'Classy News Yearly Subscription',
+            currency: 'sek'
+          )
+  
+        if charge.paid?
+          current_user.update_attribute(:role, 'subscriber')
+          render json: { message: 'Your payment was successful' }
+        else    
+          render_error(charge.errors)
+        end
+
+      rescue => error
+        render_error(error.message)
+      end
+    else
+      render_error('No stripe token detected')
+    end
+  end
+
+  def new; end
```

Comment:
do you need to have the new action defined?

Reply:
Deleted